### PR TITLE
Add Aruba fingerprints via SNMP

### PR DESCRIPTION
**Pull request by dcollado-r7**
## Description
Add new fingerprints to snmp_sysdescr.xml and snmp_sysobjid.xml for [Aruba Switch devices](https://www.arubanetworks.com/en-gb/products/switches/)

## Motivation and Context
Improved coverage

## How Has This Been Tested?
- `rake tests`
- `./bin/recog_verify  xml/snmp_sysdescr.xml`
- `./bin/recog_verify  xml/snmp_sysobjid.xml`

## Types of changes
- New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have updated the documentation accordingly (or changes are not required).
- [x] I have added tests to cover my changes (or new tests are not required).
- [x] All new and existing tests passed.

### CODE REVIEW

File: xml/snmp_sysdescr.xml [hunk 1]
```diff
@@ -749,6 +749,21 @@
...
+
+  <fingerprint pattern="^Aruba\s(JL\d+A)\s(\d+[A-Z]?)\S+\sSwitch.+ROM\s([A-Z]+(?:\.\d+)+)">
+    <description>HP Aruba Network Switch</description>
+    <example hw.product="JL256A" hw.family="2930F" hw.version="WC.16.01.0010">Aruba JL256A 2930F-48G-PoE+-4SFP+ Switch, revision WC.16.11.0004, ROM WC.16.01.0010</example>
+    <param pos="0" name="hw.device" value="Switch"/>
+    <param pos="0" name="hw.vendor" value="Aruba Networks"/>
+    <param pos="1" name="hw.product"/>
+    <param pos="2" name="hw.family"/>
+    <param pos="3" name="hw.version"/>
+    <param pos="0" name="os.vendor" value="Aruba Networks"/>
```
Comment (sdynes-r7) [inline xml/snmp_sysdescr.xml:764]: Alot of switches tend to have an OS specific to them, have you checked to see if that is the case with these arubas switches? just if it is we can infer a os.product from it too.

Comment (dcollado-r7) [reply xml/snmp_sysdescr.xml]: I can’t find Aruba documentation on this matter, might be better to play it safe and leave it out

File: xml/snmp_sysdescr.xml [hunk 2]
```diff
@@ -749,6 +749,21 @@
...
   </fingerprint>
 
+  <!--======================================================================
+                               Aruba
+  =======================================================================-->
+
+  <fingerprint pattern="^Aruba\s(JL\d+A)\s(\d+[A-Z]?)\S+\sSwitch.+ROM\s([A-Z]+(?:\.\d+)+)">
+    <description>HP Aruba Network Switch</description>
+    <example hw.product="JL256A" hw.family="2930F" hw.version="WC.16.01.0010">Aruba JL256A 2930F-48G-PoE+-4SFP+ Switch, revision WC.16.11.0004, ROM WC.16.01.0010</example>
+    <param pos="0" name="hw.device" value="Switch"/>
```
[the remark this answers is not in the document]
Comment (dcollado-r7) [reply xml/snmp_sysdescr.xml]: Reordered the properties

**Comment by mkienow-r7** [comment]
Please remove the newline at the end of the files for consistency with existing fingerprint files.

File: xml/snmp_sysdescr.xml [hunk 3]
```diff
@@ -749,6 +749,21 @@
     <param pos="0" name="os.device" value="WAP"/>
   </fingerprint>
 
+  <!--======================================================================
+                               Aruba
+  =======================================================================-->
+
+  <fingerprint pattern="^Aruba\s(JL\d+A)\s(\d+[A-Z]?)\S+\sSwitch.+ROM\s([A-Z]+(?:\.\d+)+)">
+    <description>HP Aruba Network Switch</description>
+    <example hw.product="JL256A" hw.family="2930F" hw.version="WC.16.01.0010">Aruba JL256A 2930F-48G-PoE+-4SFP+ Switch, revision WC.16.11.0004, ROM WC.16.01.0010</example>
```
[the remark this answers is not in the document]
Comment (mkienow-r7) [reply xml/snmp_sysdescr.xml:758]: I believe you used the Aruba fingerprint from `xml/dhcp_vendor_class.xml` as a model to form this new fingerprint. As a part of these updates please create a commit to update that fingerprint as well.

[the remark this answers is not in the document]
Comment (mkienow-r7) [reply xml/snmp_sysdescr.xml:758]: * `.vendor`
* `.family`
* `.product`
* `.version`
* `.build`
* `.edition`
* `.model`
* `.device`
* `.certainty`

**Comment by mkienow-r7** [comment]
Thank you for the contribution @dcollado-r7!

**Pull request merged**